networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/InternalValidationResult.java

```diff
@@ -13,8 +13,23 @@
 
 package tech.pegasys.teku.networking.eth2.gossip.topics.validation;
 
-public enum ValidationResult {
-  VALID,
-  SAVED_FOR_FUTURE,
-  INVALID
+public enum InternalValidationResult {
+  ACCEPT,
+  SAVE_FOR_FUTURE,
+  IGNORE,
+  REJECT;
+
+  public io.libp2p.core.pubsub.ValidationResult getGossipSubValidationResult() {
+    switch (this) {
+      case SAVE_FOR_FUTURE:
+      case IGNORE:
+        return io.libp2p.core.pubsub.ValidationResult.Ignore;
+      case REJECT:
+        return io.libp2p.core.pubsub.ValidationResult.Invalid;
+      case ACCEPT:
+        return io.libp2p.core.pubsub.ValidationResult.Valid;
+      default:
+        throw new UnsupportedOperationException("Enum missing value");
+    }
```

Comment:
I'd suggest adding a constructor parameter so each enum value holds it's gossip sub validation result rather than having to switch here. That way it's impossible to have an unknown mapping.

Reply:
After the class renaming you can `import io.libp2p.core.pubsub.ValidationResult`

Reply:
Both good catches. Thank you. Made the changes.